### PR TITLE
[RG-152] Enable menu bar in welcome page. show File and Help menus

### PR DESCRIPTION
**Pull request by TarasSydor**
> ### Motivate of the pull request
Given PR fulfills RG-152 ticket and enables menu bar when welcome page is shown.

> ### Describe the technical details
The idea is to 'mark' menu actions, that should be shown at the welcome page, with a special property.
When showing or hiding welcome page, we loop over all menus and their children and check that property. If it's set and welcome page is shown - action becomes visible. 

Two states of visibility were introduced: Fully and Partially visible. It can be set for parent actions, which will show all the sub items on the welcome page (e.g. Help menu or Recent projects ).

Now Welcome Page looks as follows:
image
image

### CODE REVIEW

**Review by skyler-rs — APPROVED** [review]
Implementation looks great, this logic will help protect against new actions accidentally getting added to the WelcomePage menus.

File: src/MainWindow/main_window.cpp [hunk 1]
```diff
@@ -513,10 +530,30 @@ void MainWindow::createToolBars() {
...
-  debugToolBar->setVisible(show);
+void MainWindow::updateMenusVisibility(const bool welcomePageShown) {
+  updateActionsVisibility(menuBar()->actions(), welcomePageShown);
+
+  fileToolBar->setVisible(!welcomePageShown);
+  debugToolBar->setVisible(!welcomePageShown);
+}
+
+void MainWindow::updateActionsVisibility(const QList<QAction*>& actions,
+                                         const bool welcomePageShown) {
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:541]: you don't need const here

File: src/MainWindow/main_window.cpp [hunk 2]
```diff
@@ -54,6 +54,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "Utils/FileUtils.h"
 #include "foedag_version.h"
 
+#define WELCOME_PAGE_MENU_PROP "showOnWelcomePage"
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:57]: could you please replace #define with const QString {};?

File: src/MainWindow/main_window.cpp [hunk 3]
```diff
@@ -513,10 +530,30 @@ void MainWindow::createToolBars() {
   debugToolBar->addAction(stopAction);
 }
 
-void MainWindow::showMenus(bool show) {
-  menuBar()->setVisible(show);
-  fileToolBar->setVisible(show);
-  debugToolBar->setVisible(show);
+void MainWindow::updateMenusVisibility(const bool welcomePageShown) {
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:533]: no need const here

**Review by KochynVolodymyr — APPROVED** [review]

**Pull request merged**